Makefile

```diff
@@ -119,6 +119,9 @@ crd: ensure-controller-gen ensure-yq
 	$(foreach p,$(wildcard ./config/crds/*.yaml),$(call strip-yaml-break,$(p)))
 	@echo Patching CRD files for additional static information
 	$(foreach p,$(wildcard ./config/crdspatch/*.yaml),$(call patch-crd-yq,$(subst ./config/crdspatch/,./config/crds/,$(p)),$(p)))
+	# Patch ClusterProvision CRD to remove the massive PodSpec def we consider an internal implementation detail:
+	@echo Patching ClusterProvision CRD yaml to remove overly verbose PodSpec details:
+	yq d -i config/crds/hive.openshift.io_clusterprovisions.yaml "spec.validation.openAPIV3Schema.properties.spec.properties.podSpec"
```

Comment:
can we use the existing patch (crdspatch) instead of another yq command ?

also don't delete it entirely maybe?

Reply:
The current crdspatch is a merge, I am not aware of any way to do a merge that deletes something.

What purpose would keeping podSpec in there serve?

Reply:
I can delete just .properties beneath that and get most of it but I'm not clear why we'd want to keep the podSpec at all.

Reply:
> I can delete just .properties beneath that and get most of it but I'm not clear why we'd want to keep the podSpec at all. Seems more logical that the podSpec entirely is the unknown field to be preserved, and users don't need to see it in the CRD definition.

because that shows up that there is a podSpec object, just that we don't care/know what's in it. compared to completely ghost field..

Reply:
I think I'd prefer ghost field honestly, there's no one I want seeing that field there other than hive internals, so I think the more hidden it is the better. (provided it works, which I think it does in this form)

Reply:
/lgtm